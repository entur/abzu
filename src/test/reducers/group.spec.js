/*
 *  Licensed under the EUPL, Version 1.2 or â€“ as soon they will be approved by
 the European Commission - subsequent versions of the EUPL (the "Licence");
 You may not use this work except in compliance with the Licence.
 You may obtain a copy of the Licence at:

 https://joinup.ec.europa.eu/software/page/eupl

 Unless required by applicable law or agreed to in writing, software
 distributed under the Licence is distributed on an "AS IS" basis,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the Licence for the specific language governing permissions and
 limitations under the Licence. */

import groupOfStopPlaceReducer from "./../../reducers/groupOfStopPlacesReducer";
import groupOfStopPlaceQuery from "./json/groupOfStopPlace.json";
import groupOfStopPlaceMutation from "./json/groupOfStopPlaceMutation.json";
import mapHelper from "../../modelUtils/mapToQueryVariables";

window.config = {
  defaultLanguageCode: "nor",
};

describe("Model: map Group of Stop Place from server to expected client model", () => {
  test("should map GraphQL query result to client model for Group of StopPlace", () => {
    const action = {
      type: "APOLLO_QUERY_RESULT",
      result: groupOfStopPlaceQuery,
      operationName: "getGroupOfStopPlaces",
    };

    const state = groupOfStopPlaceReducer({}, action);
    expect(state).toMatchSnapshot();
  });

  test("should map GraphQL mutation result to client model for Group of StopPlace", () => {
    const action = {
      type: "APOLLO_MUTATION_RESULT",
      result: groupOfStopPlaceMutation,
      operationName: "mutateGroupOfStopPlaces",
    };

    const state = groupOfStopPlaceReducer({}, action);
    expect(state).toMatchSnapshot();
  });

  test("should map client model for Group of StopPlace to schema", () => {
    const action = {
      type: "APOLLO_QUERY_RESULT",
      result: groupOfStopPlaceQuery,
      operationName: "getGroupOfStopPlaces",
    };
    const state = groupOfStopPlaceReducer({}, action);
    const clientGroupOfStopPlace = state.current;
    const groupOfStopPlaceExample = mapHelper.mapGroupOfStopPlaceToVariables(
      clientGroupOfStopPlace,
    );
    expect(groupOfStopPlaceExample).toMatchSnapshot();
  });
});
