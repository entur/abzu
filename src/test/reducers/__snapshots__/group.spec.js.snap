// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Model: map Group of Stop Place from server to expected client model > should map GraphQL mutation result to client model for Group of StopPlace 1`] = `
{
  "centerPosition": [
    58.955171,
    5.7400424999999995,
  ],
  "current": {
    "description": "",
    "entityType": "GROUP_OF_STOP_PLACE",
    "id": "NSR:GroupOfStopPlaces:1",
    "location": [
      58.955171,
      5.7400424999999995,
    ],
    "members": [
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:27423",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.954075,
          5.742027,
        ],
        "name": "Strømsbrua",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:47270",
            "location": [
              58.954487,
              5.742127,
            ],
            "publicCode": "",
          },
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:47271",
            "location": [
              58.953663,
              5.741927,
            ],
            "publicCode": "",
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": null,
        "tariffZones": [],
      },
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:10",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.956267,
          5.741593,
        ],
        "name": "Paradis",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:17",
            "location": [
              58.956267,
              5.741593,
            ],
            "publicCode": null,
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": "railReplacementBus",
        "tariffZones": [],
      },
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "description": null,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:26635",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.955028,
          5.738058,
        ],
        "name": "Arne Garborgs vei",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:46027",
            "location": [
              58.95505,
              5.738073,
            ],
            "publicCode": "",
          },
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:46028",
            "location": [
              58.955006,
              5.738042,
            ],
            "publicCode": "",
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": null,
        "tariffZones": [],
      },
    ],
    "name": "Gruppe2",
    "topographicPlaces": [],
  },
  "isModified": false,
  "notFound": false,
  "original": {
    "description": "",
    "entityType": "GROUP_OF_STOP_PLACE",
    "id": "NSR:GroupOfStopPlaces:1",
    "location": [
      58.955171,
      5.7400424999999995,
    ],
    "members": [
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:27423",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.954075,
          5.742027,
        ],
        "name": "Strømsbrua",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:47270",
            "location": [
              58.954487,
              5.742127,
            ],
            "publicCode": "",
          },
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:47271",
            "location": [
              58.953663,
              5.741927,
            ],
            "publicCode": "",
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": null,
        "tariffZones": [],
      },
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:10",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.956267,
          5.741593,
        ],
        "name": "Paradis",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:17",
            "location": [
              58.956267,
              5.741593,
            ],
            "publicCode": null,
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": "railReplacementBus",
        "tariffZones": [],
      },
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "description": null,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:26635",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.955028,
          5.738058,
        ],
        "name": "Arne Garborgs vei",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:46027",
            "location": [
              58.95505,
              5.738073,
            ],
            "publicCode": "",
          },
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:46028",
            "location": [
              58.955006,
              5.738042,
            ],
            "publicCode": "",
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": null,
        "tariffZones": [],
      },
    ],
    "name": "Gruppe2",
    "topographicPlaces": [],
  },
  "zoom": 15,
}
`;

exports[`Model: map Group of Stop Place from server to expected client model > should map GraphQL query result to client model for Group of StopPlace 1`] = `
{
  "centerPosition": [
    58.956267,
    5.741593,
  ],
  "current": {
    "description": "",
    "entityType": "GROUP_OF_STOP_PLACE",
    "id": "NSR:GroupOfStopPlaces:1",
    "location": [
      58.956267,
      5.741593,
    ],
    "members": [
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:10",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.956267,
          5.741593,
        ],
        "name": "Paradis",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:17",
            "location": [
              58.956267,
              5.741593,
            ],
            "publicCode": null,
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": "railReplacementBus",
        "tariffZones": [],
      },
    ],
    "name": "Gruppe",
    "topographicPlaces": [],
  },
  "isModified": false,
  "notFound": false,
  "original": {
    "description": "",
    "entityType": "GROUP_OF_STOP_PLACE",
    "id": "NSR:GroupOfStopPlaces:1",
    "location": [
      58.956267,
      5.741593,
    ],
    "members": [
      {
        "accessibilityAssessment": {
          "limitations": {
            "audibleSignalsAvailable": "UNKNOWN",
            "escalatorFreeAccess": "UNKNOWN",
            "liftFreeAccess": "UNKNOWN",
            "stepFreeAccess": "UNKNOWN",
            "visualSignsAvailable": "UNKNOWN",
            "wheelchairAccess": "UNKNOWN",
          },
        },
        "alternativeNames": [],
        "belongsToGroup": false,
        "entityType": "STOP_PLACE",
        "fareZones": [],
        "groups": [],
        "hasExpired": false,
        "id": "NSR:StopPlace:10",
        "isActive": true,
        "isFuture": false,
        "isMemberOfGroup": true,
        "location": [
          58.956267,
          5.741593,
        ],
        "name": "Paradis",
        "parking": [],
        "permanentlyTerminated": false,
        "quays": [
          {
            "accessibilityAssessment": {
              "limitations": {
                "audibleSignalsAvailable": "UNKNOWN",
                "escalatorFreeAccess": "UNKNOWN",
                "liftFreeAccess": "UNKNOWN",
                "stepFreeAccess": "UNKNOWN",
                "visualSignsAvailable": "UNKNOWN",
                "wheelchairAccess": "UNKNOWN",
              },
            },
            "boardingPositions": [],
            "description": "",
            "id": "NSR:Quay:17",
            "location": [
              58.956267,
              5.741593,
            ],
            "publicCode": null,
          },
        ],
        "stopPlaceType": "onstreetBus",
        "submode": "railReplacementBus",
        "tariffZones": [],
      },
    ],
    "name": "Gruppe",
    "topographicPlaces": [],
  },
  "zoom": 15,
}
`;

exports[`Model: map Group of Stop Place from server to expected client model > should map client model for Group of StopPlace to schema 1`] = `
{
  "description": {
    "lang": "nor",
    "value": "",
  },
  "id": "NSR:GroupOfStopPlaces:1",
  "members": [
    {
      "ref": "NSR:StopPlace:10",
    },
  ],
  "name": {
    "lang": "nor",
    "value": "Gruppe",
  },
}
`;
